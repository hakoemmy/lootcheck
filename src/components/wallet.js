import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Wallet extends Component {

    render() {
        return(
            <div>
               <h3 className='balance'>Wallet balance: {this.props.balance}</h3>
            </div>
        )
    }
}

export default connect(state => { return {balance: state} },null)(Wallet);