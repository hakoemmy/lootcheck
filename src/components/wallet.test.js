import React from 'react';
import {shallow} from 'enzyme';
import { Wallet } from './wallet';
import '../setupTests';

describe('Wallet component', () => {
    const props = { balance: 20};
    
    const wallet = shallow(<Wallet {...props}/>);
    it ('renders properly', () => {
       expect(wallet).toMatchSnapshot();
    });
    it('displays balance from props',() => {
        expect(wallet.find('.balance').text()).toEqual('Wallet balance: 20');
    })
});