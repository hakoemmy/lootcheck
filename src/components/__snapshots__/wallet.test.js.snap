// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet component renders properly 1`] = `ShallowWrapper {}`;
